File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.cs
using System;

public class Solution 
{
    public int[] solution(int[] numbers, string direction) 
    {
        int nl = numbers.Length;
        int lastTemp = numbers[nl-1];
        int firstTemp = numbers[0];
        switch(direction)
        {
            case "right":
                for(int i=0; i<nl-1; i++)
                {
                    numbers[nl-1-i] = numbers[nl-2-i];
                }
                numbers[0] = lastTemp;
                break;
            case "left":
                for(int i=1; i<nl; i++)
                {
                    numbers[i-1] = numbers[i];
                }
                numbers[nl-1] = firstTemp;  
                break;
        }
        return numbers;
    }
}

File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> numbers, string direction)
{
        if(direction == "right")
        {           
            int temp = numbers[numbers.size()-1];
            for(int i=numbers.size()-1; i>=0; i--)
            {
                numbers[i] = numbers[i-1];
            }
            numbers[0] = temp;
        }  
        else if(direction == "left")
        {       
            int temp = numbers[0];
            for(int i=0; i<numbers.size()-1; i++)
            {
                numbers[i] = numbers[i+1];
            }
            numbers[numbers.size()-1] = temp;
        }
    return numbers;
}